lib/grplatt.gi

```diff
@@ -3654,6 +3654,586 @@ function(G)
+        UZ := Image(m2,U);
+        VZ := Image(lambda,UZ);
+        VZrep := PreImagesRepresentative(m2,VZ);
+        if Size(C)=1 then return VZrep;fi;
+        VZ := RightCoset(C,VZrep);
+        for V in VZ do
+            if Order(V) = p then return V; fi;
+        od;
+        Error("No element found");
+    end;
+
+    # Calculating AutSl
+    AutSlgens := [];
+    for lambda in AutpSgens do
+        alphaSlgen := List(Slgen,U -> m2lambm2inv(U,lambda,m2,C,p));
+        alpha := GroupHomomorphismByImagesNC(Sldq,Sldq,Slgen,alphaSlgen);
+        Add(AutSlgens,alpha);
+    od;
+
+    # Checking if AutSlgens contains an automorphism that is not a
+    # conjugator automorphism times a field automorphism.
+    scalars := Filtered(GF(q), x -> x^d = One(GF(q)));
+    scalars := SmallGeneratingSet(AsGroup(scalars));
+    scalarMatrices := List(scalars,x -> x * IdentityMatrix(GF(q),d));
+    f := FrobeniusAutomorphism(GF(q));
+    flisinit := List([1..Order(f)],x->f^x);
+    finvlisinit := List(flisinit,f -> f^-1);
+    imagesOfscalars := List(flisinit,f -> List(scalars,f));
+    MakeFieldAutomorphismOfSldq := function(f)
+      local row,M,applyfonrow,applyf,fgens,x;
+      applyfonrow := row -> List(row,x -> Image(f,x));
+      applyf := M -> List(M,applyfonrow);
+      fgens := List(Slgen,applyf);
+      return GroupHomomorphismByImagesNC(Sldq,Sldq,Slgen,fgens);
+    end;
+    finvmatrixlist := List(finvlisinit,MakeFieldAutomorphismOfSldq);
+    if type="PSL" or type="PSp" then
+        for alpha in AutSlgens do
+          conj := false;
+          FirstOfimagesOfscalarMatrices := List(scalarMatrices,M -> alpha(M)[1][1]);
+          findlis := Filtered([1..Length(flisinit)], i -> FirstOfimagesOfscalarMatrices = imagesOfscalars[i]);
+          finvlis := List(findlis,i -> finvmatrixlist[i]);
+          for finv in finvlis do
+            alphafinv := alpha * finv;
+            if IsConjugatorAutomorphism(alphafinv) then
+              conj := true;
+              break;
+            fi;
+          od;
+          if not conj then return false; fi;
+        od;
+    elif type="POmega+" then
+        O := GO(1,d,q);
+        C := Center(GO(1,d,q));
+        phi := NaturalHomomorphismByNormalSubgroupNC(O,C);
+        PO := Image(phi);
+        POgens := GeneratorsOfGroup(PO);
+        POgens := List(POgens,x -> PreImagesRepresentative(phi,x));
+        POgens := List(POgens,x-> ConjugatorAutomorphismNC(Sldq,x));
+        PO := GroupWithGenerators(POgens);
+        if d > 8 then
+            for alpha in AutSlgens do
+                if not alpha in PO then return false; fi;
+            od;
+        elif d = 8 then
+
+        fi;
+    fi;
+    return true;
+end);
+
+##################################################################################
+##
+#F  MinimalFaithfulPermutationDegreeOfAlmostSimpleGroupWithSimpleSubgroup(<A>,<S>)
+##
+##  Returns mu(A) where S < A < Aut(S) for simple group S
+##  if it can compute it easily, else returns -1.
```

Comment:
I'm not sure what  the purpose of returning -1 here is. When the function gets called (in MinimalFaithfulPermutationDegreeOfSemiSimpleGroup), it immediately tests for value -1 and then doe the extra work of determining the correct degree. I would prefer to move this determination here, into this function, making is a useful general-purpose function. (This would mean that `MinimalFaithfulPermutationDegreeOfSemiSimpleGroup` becomes a very easy function, indeed one that could easily be integrated into the general `MinimalFaithfulPermutationDegree` routine.

Reply:
I was trying to keep this function as fast as possible, mainly for testing purposes. But I'll move the determination here, as you say.